cpu/sam0_common/periph/spi.c

```diff
@@ -82,8 +106,23 @@ void spi_init(spi_t bus)
      * no synchronization needed, as SERCOM device is not enabled */
     dev(bus)->CTRLB.reg = (SERCOM_SPI_CTRLB_CHSIZE(0) | SERCOM_SPI_CTRLB_RXEN);
 
+#ifdef MODULE_PERIPH_DMA
+    if (_use_dma(bus)) {
+        _dma_state[bus].rx_dma = dma_acquire_channel();
+        _dma_state[bus].tx_dma = dma_acquire_channel();
+        dma_setup(_dma_state[bus].tx_dma,
+                  spi_config[bus].tx_trigger, 0, false);
+        dma_setup(_dma_state[bus].rx_dma,
+                  spi_config[bus].rx_trigger, 1, true);
+        dma_prepare(_dma_state[bus].rx_dma, DMAC_BTCTRL_BEATSIZE_BYTE_Val,
+                    (void*)&dev(bus)->DATA.reg, NULL, 1, 0);
```

Comment:
You can drop the cast.

Reply:
Unfortunately not, this cast removes the `volatile` qualifier from `DATA.reg`. It doesn't compile without the cast:
```
/home/koen/dev/RIOT-dma/cpu/sam0_common/periph/spi.c: In function 'spi_init':
/home/koen/dev/RIOT-dma/cpu/sam0_common/periph/spi.c:118:21: error: passing argument 3 of 'dma_prepare' discards 'volatile' qualifier from pointer target type [-Werror=discarded-qualifiers]
  118 |                     &dev(bus)->DATA.reg, NULL, 1, 0);
      |                     ^~~~~~~~~~~~~~~~~~~
In file included from /home/koen/dev/RIOT-dma/cpu/samd21/include/periph_cpu.h:25,
                 from /home/koen/dev/RIOT-dma/drivers/include/periph/spi.h:74,
                 from /home/koen/dev/RIOT-dma/cpu/sam0_common/periph/spi.c:31:
/home/koen/dev/RIOT-dma/cpu/sam0_common/include/periph_cpu_common.h:769:56: note: expected 'const void *' but argument is of type 'volatile uint32_t *' {aka 'volatile long unsigned int *'}
  769 | void dma_prepare(dma_t dma, uint8_t width, const void *src, void *dst,
      |                                            ~~~~~~~~~~~~^~~
```